### PR TITLE
Completed week 4 challenges

### PR DESCRIPTION
**Pull request by adeolaidowu**
Kindly review my solution

### CODE REVIEW

File: src/count-islands/index.js [hunk 1]
```diff
@@ -1,3 +1,38 @@
-function countIslands(grid) {}
+function countIslands(grid) {
+  const rowLength = grid.length;
+  const colLength = grid[0].length;
+  let islands = 0;
+  const search = (currentRow, currentCol) => {
```
Comment (flinchy) [inline src/count-islands/index.js:5]: move the search() function out of the countIsLands() function. A function should have a single responsibility that way your code is easily maintainable.

File: src/word-search/index.js [hunk 2]
```diff
@@ -1,3 +1,53 @@
-function wordSearch(words, word) {}
+function wordSearch(words, word) {
+  let wordArr;
+  let wordsCopy;
+  let pathFound = false;
+  const rowLength = words.length;
+  const colLength = words[0].length;
+
+  const searchAdjacent = (currentRow, currentCol) => {
```
Comment (flinchy) [inline src/word-search/index.js:8]: move the searchAdjacent() function out of wordSearch() function

**Comment by adeolaidowu** [comment]
Thanks for the feedback

**Comment by flinchy** [comment]
you are welcome

**Pull request merged**